src/lib/auth/guards/authentication.guard.ts

```diff
@@ -13,21 +12,43 @@ export class VantageAuthenticationGuard implements CanActivate {
   constructor(private _router: Router, private _sessionService: VantageSessionService) {}
 
   canActivate(next: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<boolean> {
-    return this._sessionService
-      .getInfo()
-      .pipe(timeout(5000))
-      .pipe(
-        catchError((e: HttpErrorResponse) => {
-          // if not logged in, go ahead and log in...otherwise logout
-          // append the current path so we get redirected back upon login
-          e.status === UNAUTHORIZED
-            ? (window.location.href = '/start-login?nonce=' + Math.floor(1000000000 + Math.random() * 9000000000))
-            : this._sessionService.logout();
-          throw e;
-        }),
-        map(() => {
-          return true;
-        }),
-      );
+    showPreLoader();
+    return this._sessionService.getInfo().pipe(
+      timeout(5000),
+      catchError((e: HttpErrorResponse) => {
+        // if not logged in, go ahead and log in...otherwise logout
+        // append the current path so we get redirected back upon login
+        e.status === UNAUTHORIZED
+          ? (window.location.href = '/start-login?nonce=' + Math.floor(1000000000 + Math.random() * 9000000000))
+          : this._sessionService.logout();
+        throw e;
+      }),
+      map(() => true),
+      finalize(() => hidePreLoader()),
+    );
+  }
+}
+
+// PRE_LOADER_COUNT variable is defined inside pre-loader.html
+
+function showPreLoader(): void {
```

Comment:
I'm thinking we should move these functions into some other place...

Reply:
Agreed, thoughts on where?

Reply:
Ideally outside auth package, but ng-packagr build issues incoming in 5 4 3 2....

Reply:
Well if we figure out https://github.com/Teradata/vantage-ui-platform/issues/39, then this should be possible too.

Reply:
I was thinking a pre-loader package, but we want this guard to still work even if the pre-loader is not used?

Reply:
Even if we call the functions, wouldnt the guard just work either way?

Reply:
Moved into pre-loader package.